addons/sale/models/sale_order_line.py

```diff
@@ -134,35 +134,20 @@ def _compute_price_reduce_taxexcl(self):
         for line in self:
             line.price_reduce_taxexcl = line.price_subtotal / line.product_uom_qty if line.product_uom_qty else 0.0
 
+    @api.depends('product_id')
     def _compute_tax_id(self):
         for line in self:
             line = line.with_company(line.company_id)
-            fpos = line.order_id.fiscal_position_id or line.order_id.fiscal_position_id.get_fiscal_position(line.order_partner_id.id)
+            fpos = line.order_id.fiscal_position_id or line.order_id.fiscal_position_id._get_fiscal_position(line.order_partner_id)
```

Comment:
```suggestion
            fpos = line.order_id.fiscal_position_id
```

Ideally, this fallback should be removed, it makes no sense to apply a magic hidden fpos to the line taxes when it's not clearly specified on the SO level :thinking: